lib/src/expression/micro/parser.dart

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2016, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:angular_ast/angular_ast.dart';
+import 'package:angular_ast/src/expression/micro/ast.dart';
+import 'package:angular_ast/src/expression/micro/lexer.dart';
+import 'package:angular_ast/src/expression/micro/token.dart';
+import 'package:meta/meta.dart';
+
+class NgMicroParser {
+  @literal
+  const factory NgMicroParser() = NgMicroParser._;
+
+  const NgMicroParser._();
+
+  NgMicroAst parse(
+    String directive,
+    String expression, {
+    @required String sourceUrl,
+  }) {
+    final tokens = const NgMicroLexer().tokenize(expression).iterator;
+    return new _RecursiveMicroAstParser(
+      directive,
+      expression,
+      sourceUrl,
+      tokens,
+    )
+        .parse();
+  }
+}
+
+class _RecursiveMicroAstParser {
+  final String _directive;
+  final String _expression;
+  final String _sourceUrl;
+  final Iterator<NgMicroToken> _tokens;
+
+  final references = <ReferenceAst>[];
+  final properties = <PropertyAst>[];
+
+  _RecursiveMicroAstParser(
+    this._directive,
+    this._expression,
+    this._sourceUrl,
+    this._tokens,
+  );
+
+  NgMicroAst parse() {
+    while (_tokens.moveNext()) {
+      final token = _tokens.current;
+      if (token.type == NgMicroTokenType.letKeyword) {
+        _parseLet();
+      } else if (token.type == NgMicroTokenType.bindIdentifier) {
+        _parseBind();
+      } else if (token.type != NgMicroTokenType.endExpression) {
+        throw _unexpected(token);
+      }
+    }
+    return new NgMicroAst(assignments: references, properties: properties);
+  }
+
+  void _parseBind() {
+    final name = _tokens.current.lexeme;
+    if (!_tokens.moveNext() ||
+        _tokens.current.type != NgMicroTokenType.bindExpressionBefore ||
+        !_tokens.moveNext() ||
+        _tokens.current.type != NgMicroTokenType.bindExpression) {
+      throw _unexpected();
+    }
+    final value = _tokens.current.lexeme;
+    properties.add(new PropertyAst(
+      '${_directive}${name[0].toUpperCase()}${name.substring(1)}',
+      new ExpressionAst.parse(
+        value,
+        sourceUrl: _sourceUrl,
+      ),
+    ));
+  }
+
+  void _parseLet() {
+    String identifier;
+    if (!_tokens.moveNext() ||
+        _tokens.current.type != NgMicroTokenType.letKeywordAfter ||
+        !_tokens.moveNext() ||
+        _tokens.current.type != NgMicroTokenType.letIdentifier) {
+      throw _unexpected();
```

Comment:
Let's chat next week - I'll leave this as-is for now, and let's come up with a strategy that will allow format exceptions to be reported without blocking analysis or tooling.